Documentation/Books/Cookbook/AQL/Joins.md

```diff
@@ -359,194 +359,209 @@ arangosh> db.written.save("authors/2938210813",
-          "_to" : "books/2980088317", 
-          "_rev" : "3006237181", 
-          "_key" : "3006237181", 
-          "pages" : "1-10" 
-        }, 
-        "vertex" : { 
-          "_id" : "authors/2935261693", 
-          "_rev" : "2935261693", 
-          "_key" : "2935261693", 
-          "name" : { 
-            "first" : "John", 
-            "last" : "Doe" 
-          } 
-        } 
-      }, 
-      { 
-        "edge" : { 
-          "_id" : "written/3012856317", 
-          "_from" : "authors/2938210813", 
-          "_to" : "books/2980088317", 
-          "_rev" : "3012856317", 
-          "_key" : "3012856317", 
-          "pages" : "11-20" 
-        }, 
-        "vertex" : { 
-          "_id" : "authors/2938210813", 
-          "_rev" : "2938210813", 
-          "_key" : "2938210813", 
-          "name" : { 
-            "first" : "Maxima", 
-            "last" : "Musterfrau" 
-          } 
-        } 
-      } 
-    ] 
-  } 
+...> "FOR b IN books " +
+...> "LET authorsByBook = ( " +
+...> "    FOR author IN INBOUND b._id written" +
+...> "    RETURN DISTINCT author " +
+...> ") " +
+...> "RETURN { " +
+...> "    book: b, " +
+...> "    authors: authorsByBook " +
+...> "} "
+...> ).toArray();
+[
+  {
+    "book" : {
+      "_key" : "2980088317",
+      "_id" : "books/2980088317",
+      "_rev" : "2980088317",
+      "title" : "The beauty of JOINS"
+    },
+    "authors" : [
+      {
+        "_key" : "2938210813",
+        "_id" : "authors/2938210813",
+        "_rev" : "2938210813",
+        "name" : {
+          "first" : "Maxima",
+          "last" : "Musterfrau"
+        }
+      },
+      {
+        "_key" : "2935261693",
+        "_id" : "authors/2935261693",
+        "_rev" : "2935261693",
+        "name" : {
+          "first" : "John",
+          "last" : "Doe"
+        }
+      }
+    ]
+  }
 ]
 ```
 
-Or if you want to hide the information stored in the edge.
+Or if you also want the information stored in the edges.
 
 ```json
 arangosh> db._query(
-........>"FOR b IN books RETURN {" +
-........>"     book: b, authors: " +
-........>"          NEIGHBORS(books, written, b._id, 'inbound')[*].vertex }"
-........>).toArray();
-[ 
-  { 
-    "book" : { 
-      "title" : "The beauty of JOINS", 
-      "_id" : "books/2980088317", 
-      "_rev" : "2980088317", 
-      "_key" : "2980088317" 
-    }, 
-    "authors" : [ 
-      { 
-        "_id" : "authors/2935261693", 
-        "_rev" : "2935261693", 
-        "_key" : "2935261693", 
-        "name" : { 
-          "first" : "John", 
-          "last" : "Doe" 
-        } 
-      }, 
-      { 
-        "_id" : "authors/2938210813", 
-        "_rev" : "2938210813", 
-        "_key" : "2938210813", 
-        "name" : { 
-          "first" : "Maxima", 
-          "last" : "Musterfrau" 
-        } 
-      } 
-    ] 
-  } 
+...> "FOR b IN books " +
+...> "LET authorsByBook = ( " +
+...> "    FOR author, writtenBy IN INBOUND b._id written " +
+...> "    RETURN DISTINCT { " +
```

Comment:
No need for DISTINCT here. Each edge of an author will be visited only once.

`b._id` could be replaced by `b`.